File: backend/MenuService/DF.MenuService.Contracts/Models/Request/CreateDishRequest.cs
using DF.MenuService.Domain.Entities;
using Microsoft.AspNetCore.Http;

namespace DF.MenuService.Contracts.Models.Request;

public record CreateDishRequest(
    Guid UserId,
    Guid? MenuId, 
    string Name, 
    string? Description, 
    decimal Price, 
    Category Category, 
    IFormFile? Image,
    int CookingTime,
    List<CreateIngredientRequest> Ingredients);
    
public record UpdateDishRequest(
    Guid DishId,
    Guid? MenuId, 
    string Name, 
    string? Description, 
    decimal Price, 
    Category Category, 
    int CookingTime,
    IFormFile? Image);

File: frontend/food-delivery-platform/src/models/requests/dish/CreateDishRequest.ts
import {CreateIngredientRequest} from "./CreateIngredientRequest";
import { Category } from "../../../constants/category";

export interface CreateDishRequest {
    businessId: string          // Guid
    name: string
    description?: string | null
    price: number               // decimal
    category: Category
    image?: File | null         // IFormFile?
    cookingTime: number
    ingredients: CreateIngredientRequest[]
}
